Update kubectl create configmap section (#18885)
kubernetes/website: content/en/docs/tasks/configure-pod-container/configure-pod-configmap.md

### Create a ConfigMap Using kubectl create configmap

Use the `kubectl create configmap` command to create ConfigMaps from [directories](#create-configmaps-from-directories), [files](#create-configmaps-from-files), or [literal values](#create-configmaps-from-literal-values):

```shell
kubectl create configmap <map-name> <data-source>
```

> Where \<map-name> is the name you want to assign to the ConfigMap and \<data-source> is the directory, file, or key and literal value to draw the data from.

Reviewer: don't change the 'where' word

Reviewer: why add 'key and'?

Author: @tengqm Thanks for the review and catching that. I have made the suggested changes.

> When you are creating a ConfigMap based on a file, the key in the \<data-source> will default to the basename of the file, and the value will default to the file content.

Reviewer 2: nit:
```suggestion
When you are creating a ConfigMap based on a file, the key in the \<data-source> defaults to the basename of the file, and the value defaults to the file content.
```

Author: @sftim Totally agree. Somehow I keep making that mistake and I hope to get better.

> When you are creating a ConfigMap based on a directory, each file whose basename is a valid key in the directory will be packaged into the ConfigMap. Any directory entries except regular files are ignored (e.g. subdirectories, symlinks, devices, pipes, etc).

Reviewer 2: How about:
```suggestion
When you are creating a ConfigMap based on a directory, kubectl identifies files whose basename is a valid key in the directory and packages each of those files into the new ConfigMap. Any directory entries except regular files are ignored (e.g. subdirectories, symlinks, devices, pipes, etc).
```

Author: Done.

> When you are creating a ConfigMap based on a directory, kubectl identifies files whose basename is a valid key in the directory and packages each of those files into the new ConfigMap. Any directory entries except regular files are ignored (e.g. subdirectories, symlinks, devices, pipes, etc).

Reviewer 2: Does this fit better in the section headed “Create ConfigMaps from directories”? (I think it does).

I can see that it's important to let readers know that there's more detail to read up on if they do this.

Author: It does. My original intent was to rewrite these paragraphs so that they are easier to read. If we move it to the "Create ConfigMaps from directories" should we also move the paragraph about files to the relevant file section as well just to be consistent.

Author: I have made the changes.

kubectl create configmap game-config --from-file=configure-pod-container/configmap/
```

The above command packages each file, in this case, `game.properties` and `ui.properties` in the `configure-pod-container/configmap/` directory into the game-config ConfigMap. You can display details of the ConfigMap using the following command:

```shell
kubectl describe configmaps game-config
```

The output is similar to this:
```
Name:         game-config
Namespace:    default
Labels:       <none>
Annotations:  <none>

Data
====
game.properties:
----
enemies=aliens
lives=3
enemies.cheat=true
enemies.cheat.level=noGoodRotten
secret.code.passphrase=UUDDLRLRBABAS
secret.code.allowed=true
secret.code.lives=30
ui.properties:
----
color.good=purple
color.bad=yellow
allow.textmode=true

> how.nice.to.look=fairlyNice

Reviewer 2: “Create ConfigMaps from files” also has sample output; should that change too?

Author: Done.
